WBooks/src/app/screens/BookDetails/components/BookComments/index.js

```diff
@@ -0,0 +1,41 @@
+import React, { PureComponent } from 'react';
+import { TouchableOpacity, Text } from 'react-native';
+
+import comments from '../../../../../constants/comments';
+
+import CommentItem from './components/CommentItem';
+import styles from './styles';
+
+const INITIAL_SHOWN_COMMENTS = 2;
+
+class BookComments extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showAll: false
+    };
+  }
+
+  renderComment = item => <CommentItem key={this.elementKeyExtractor(item)} comment={item} />;
+
+  elementKeyExtractor = item => item.id.toString();
+
+  handleShowAllComments = () => this.setState({ showAll: true });
+
+  render() {
+    const { showAll } = this.state;
+    const selectedComments = showAll ? comments : comments.slice(0, INITIAL_SHOWN_COMMENTS);
```

Comment:
okay. This is not a good practice to do array things inside the render function but this will be replaced for better practices using redux 😄 Just keep it in mind.

Reply:
Where should I do this?

Reply:
a solution could be use `componentDidUpdate` getting the initial comments prop and setting a partialComments state. 
https://es.reactjs.org/docs/react-component.html#shouldcomponentupdate